cmd/estuary-shuttle/rpc.go

```diff
@@ -300,22 +297,35 @@ func (d *Shuttle) handleRpcTakeContent(ctx context.Context, cmd *drpc.TakeConten
 	defer d.addPinLk.Unlock()
 
 	for _, c := range cmd.Contents {
-		var count int64
-		err := d.DB.Model(Pin{}).Where("content = ?", c.ID).Limit(1).Count(&count).Error
+		var pin Pin
+		err := d.DB.First(&pin, "content = ?", c.ID).Error
 		if err != nil {
-			return err
-		}
-
-		// if this content is already in this shuttle, ignore pinning it
-		if count > 0 {
-			continue
+			if errors.Is(err, gorm.ErrRecordNotFound) {
+				go func(c drpc.ContentFetch) {
+					if err := d.addPin(ctx, c.ID, c.Cid, c.UserID, c.Peers, true); err != nil {
+						log.Errorf("failed to pin takeContent: %d", c.ID)
+					}
+				}(c)
+			} else {
+				log.Errorf("error finding pin: %s", err)
+				return err
+			}
 		}
 
-		go func(c drpc.ContentFetch) {
-			if err := d.addPin(ctx, c.ID, c.Cid, c.UserID, c.Peers, true); err != nil {
-				log.Errorf("failed to pin takeContent: %d", c.ID)
+		if pin.Active {
+			// if this content is already in this shuttle, resend pin complete
+			if err := d.resendPinComplete(ctx, pin); err != nil {
+				log.Error(err)
 			}
-		}(c)
+		}
+		if pin.Failed {
```

Comment:
we do not retry failed pins, instead, send a pin failure RPC message, so Estuary can mark the content as failed (as it should have been in the first place, probably did not have the right state due to dropped RPC message) and remove this content from the zones